google_api_lib/drive.py

```diff
@@ -0,0 +1,245 @@
+import httplib2
+import logging
+import os
+
+from django.conf import settings
+
+from googleapiclient import _auth
+
+from celery import shared_task
+from .utils import GoogleApiClientTask
+from puzzles.models import Puzzle
+
+logger = logging.getLogger(__name__)
+
+
+@shared_task(base=GoogleApiClientTask, bind=True)
```

Comment:
I think the convention is to collect all delayed tasks in a `/tasks.py` file (one per django app) so they're easier to find

Reply:
Hmm yeah I actually wasn't sure what the best thing to do was b/c I had to split the tasks between drive.py, which depends on Puzzle model to update google sheets url, and whitelist.py which is called in settings.py before Puzzle model exists yet to get the whitelisted emails. The whitelisted emails should really be removed from there and be made a field of a hunt. I was gonna wait to do that later, but maybe I should just do it as apart of this....

Reply:
Aha yeah. If the one in whitelist.py is needed at startup then is there much point making it asynchronous to begin with? Maybe we can just leave it synchronous for now and move the other tasks to task.py? We definitely need to improve the permissions modeling at some point but it feels it can wait for now

Reply:
It's still being called synchronously there. That's a good point though, I'll rename drive.py into task.py and either keep whitelist.py or rename that to sync_task.py or something lol.